manifests/config.pp

```diff
@@ -1,22 +1,32 @@
 class winlogbeat::config {
-
-  $registry_file = $winlogbeat::registry_file
-  $event_logs    = $winlogbeat::event_logs_final
-
-  $winlogbeat_config = {
-    'output'     => $winlogbeat::outputs,
-    'shipper'    => $winlogbeat::shipper,
-    'logging'    => $winlogbeat::logging,
-    'runoptions' => $winlogbeat::run_options,
-  }
+  $winlogbeat_config = delete_undef_values({
+    'beat_name'         => $winlogbeat::beat_name,
+    'tags'              => $winlogbeat::tags,
+    'queue_size'        => $winlogbeat::queue_size,
+    'max_procs'         => $winlogbeat::max_procs,
+    'fields'            => $winlogbeat::fields,
+    'fields_under_root' => $winlogbeat::fields_under_root,
+    'winlogbeat'        => {
+      'registry_file' => $winlogbeat::registry_file,
+      'metrics'       => $winlogbeat::metrics,
+      'event_logs'    => $winlogbeat::event_logs_final,
+    },
+    'output'            => $winlogbeat::outputs,
+    'shipper'           => $winlogbeat::shipper,
+    'logging'           => $winlogbeat::logging,
+    'runoptions'        => $winlogbeat::run_options,
+  })
 
   case $::kernel {
     'Windows' : {
+      $winlogbeat_path = 'c:\Program Files\Winlogbeat\winlogbeat.exe'
```

Comment:
Probably. I copied from the filebeat module, so it should probably be updated there as well. I had that thought but was lazy.